### PR TITLE
Sonarqube integration

### PR DESCRIPTION
**Pull request by pazone**
## What is the problem this PR solves?
Enables sonarqube code scanning and coverage reports. 

[Sonar project for fleet-server](https://sonar.elastic.dev/dashboard?id=elastic_fleet-server_AYpL8BsVaV3I-igkX4hx) is already created.

This configuration provides:

Pull request analysis: Analyses newly added code for a particular pull request
SonarQube automatically adds a comment with digest information. If no java files were changed in a pull request we will see “No coverage/duplication information” labels

Branch analysis: Runs on branches like main and analyzes the entire codebase. Associates received code coverage with the branch

## Sonarqube Requires a quality profile 
Quality profile is configured on the sonarqube side. 
Questions to the team:
- What is the desired code coverage for the project?
- Minimal code coverage for pull request? (SonarQube has a separate measure for newly added/changed code)

## Checklist
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/fleet-server#changelog)

## Related issues

- Closes https://github.com/elastic/ingest-dev/issues/2281
-->

### CODE REVIEW

**Comment by jlind23** [comment]
@pazone our buildkite build seems to be failing due to 
`2023-08-31 15:13:34 FATAL  Failed to upload and process pipeline: Pipeline upload rejected: Detected dependency cycle: :sonarqube: Continuous Code Inspection → :sonarqube: Continuous Code Inspection`

**Comment by pazone** [comment]
@jlind23 Do I understand correctly that now we don't need the `Code coverage build` stage made by Github actions?
https://github.com/elastic/fleet-server/blob/main/.github/workflows/build.yml

**Review by jlind23 — APPROVED** [review]

**Review by mrodm — COMMENTED** [review]
LGTM, just one doubt about when this new step needs to run

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -41,6 +41,18 @@ steps:
...
+      - label: ":sonarqube: Continuous Code Inspection"
+        env:
+          VAULT_SONAR_TOKEN_PATH: "kv/ci-shared/platform-ingest/elastic/fleet-server/sonar-analyze-token"
+        agents:
+          image: "docker.elastic.co/cloud-ci/sonarqube/buildkite-scanner:latest"
+        command:
+          - "buildkite-agent artifact download build/coverage.out ."
+          - "/scan-source-code.sh"
+        depends_on:
+          - "unit-test"
```
[the remark this answers is not in the document]
Comment (pazone) [reply .buildkite/pipeline.yml:55]: All branches. We also need to analyse main

**Pull request merged**